docs/dev/Pagination-v2.md

```diff
@@ -285,3 +285,87 @@ OpenSearchExecutionEngine->>+ProjectOperator: getTotalHits
     ResourceMonitorPlan-->>-ProjectOperator: value
   ProjectOperator-->>-OpenSearchExecutionEngine: value
 ```
+
+### Plan Tree changes
+
+#### Abstract Plan tree
+
+Changes:
```

Comment:
Update to `Changes to plan tree for Initial Query Request with pagination` (or similar):
Add a section to `Changes without pagination`: no changes and pageSize will be empty/null.

Reply:
We probably need a summary above for 'why' we need pageSize and the Paginate unresolved node.